### PR TITLE
Breadcrumbs unification

### PR DESCRIPTION
**Pull request by jhadvig**
@spadgett @jwforres as we discussed here is the PR unifying the usage of the breadcrumb directive. 
PTAL

### CODE REVIEW

File: assets/app/scripts/controllers/create/createFromImage.js [hunk 1]
```diff
@@ -22,6 +22,21 @@ angular.module("openshiftConsole")
 
     $scope.projectName = $routeParams.project;
     $scope.sourceURLPattern = SOURCE_URL_PATTERN;
+    $scope.imageName = $routeParams.imageName;
```
Comment (spadgett) [inline assets/app/scripts/controllers/create/createFromImage.js]: Adding values to `$scope` is a signal to me that it's used in the view. I usually don't add items there unless I need to. I'd prefer using a local var otherwise.

```
var imageName = $routeParams.imageName;
```

?

File: assets/app/scripts/controllers/create/createFromImage.js [hunk 2]
```diff
@@ -22,6 +22,21 @@ angular.module("openshiftConsole")
...
+
+    $scope.breadcrumbs = [
+      {
+        title: $scope.projectName,
```
Comment (spadgett) [inline assets/app/scripts/controllers/create/createFromImage.js]: We need to update the title to the project display name under `ProjectsService.get()` in each of these controllers. See

https://github.com/openshift/origin/blob/master/assets/app/scripts/controllers/setLimits.js#L63-L64

File: assets/app/scripts/controllers/create/nextSteps.js [hunk 3]
```diff
@@ -16,24 +16,51 @@ angular.module("openshiftConsole")
     $scope.emptyMessage = "Loading...";
     $scope.alerts = [];
     $scope.loginBaseUrl = DataService.openshiftAPIBaseUrl();
-    $scope.projectName = $routeParams.project;
     $scope.buildConfigs = {};
 
+    $scope.projectName = $routeParams.project;
+    $scope.imageName = $routeParams.imageName;
+    $scope.imageTag = $routeParams.imageTag;
+    $scope.namespace = $routeParams.namespace;
```
Comment (spadgett) [inline assets/app/scripts/controllers/create/nextSteps.js]: Same comment as above about scope vars.

File: assets/app/scripts/controllers/create/createFromImage.js (hunk 2 above)
Comment (jhadvig) [inline assets/app/scripts/controllers/create/createFromImage.js]: Why is this necessary since the route is already containing the name ? Just want to understand the point :)

Comment (spadgett) [inline assets/app/scripts/controllers/create/createFromImage.js]: It's showing the name, but not the display name. My project "hello" might have display name "Hello Openshift." The breadcrumb should show "Hello Openshift." We don't have that until after `ProjectsService.get()`.

Comment (jhadvig) [inline assets/app/scripts/controllers/create/createFromImage.js]: ou... thanks for letting me know ! :) will update

**Comment by jhadvig** [comment]
@spadgett comments addressed. PTAL :)

File: assets/app/scripts/controllers/create/nextSteps.js [hunk 4]
```diff
@@ -16,21 +16,53 @@ angular.module("openshiftConsole")
...
     $scope.buildConfigs = {};
 
+    $scope.projectName = $routeParams.project;
+    imageName = $routeParams.imageName;
+    imageTag = $routeParams.imageTag;
+    namespace = $routeParams.namespace;
+
+    var imageName = $routeParams.imageName;
+    var imageTag = $routeParams.imageTag;
+    var namespace = $routeParams.namespace;
```
Comment (spadgett) [inline assets/app/scripts/controllers/create/nextSteps.js]: Looks like this is repeated twice.

File: assets/app/scripts/controllers/create/nextSteps.js [hunk 5]
```diff
@@ -16,21 +16,53 @@ angular.module("openshiftConsole")
...
+      {
+        title: "Add to Project",
+        link: "project/" + $scope.projectName + "/create"
+      },
+      {
+        title: name,
+        link: nameLink
+      },
+      {
+        title: "Next Steps"
```
Comment (spadgett) [inline assets/app/scripts/controllers/create/nextSteps.js]: This is a little different than the previous behavior that said "Creating" if all tasks weren't complete. I think I'm OK with that, though.

**Comment by jhadvig** [comment]
@spadgett sorry for the dup., I've missed that one. Thank you

**Comment by spadgett** [comment]
LGTM [merge]

**Comment by spadgett** [comment]
@jhadvig bindata conflict.

**Comment by jhadvig** [comment]
@spadgett I've updated the PR. Needed to re-install the assets after your change was merged yesteday

**Comment by spadgett** [comment]
[merge]
